lib/jormungandr/src/Cardano/Wallet/Jormungandr/Network.hs

```diff
@@ -163,11 +181,60 @@ mkNetworkLayer j = NetworkLayer
 
     }
   where
+    getTipId' = (getTipId j) `mappingError`
+        ErrNetworkTipNetworkUnreachable
+    getBlock' t = (getBlock j t) `mappingError` \case
+        ErrGetBlockNotFound _ ->
+            ErrNetworkTipNotFound
+        ErrGetBlockNetworkUnreachable e ->
+            ErrNetworkTipNetworkUnreachable e
+
     mappingError = flip withExceptT
 
     tailOrEmpty [] = []
     tailOrEmpty (_:xs) = xs
 
+{-------------------------------------------------------------------------------
+                                 State munching
+-------------------------------------------------------------------------------}
+
+-- | A list of blocks and their hashes.
+-- The first block in this sequence is the network tip.
+-- The last block in this sequence is the block of depth _k_,
+-- which is the last unstable block.
+newtype UnstableBlocks = UnstableBlocks
```

Comment:
side-note: do we need to keep the whole block in memory? Aren't we fine keeping only the header hash :thinking:  ?

Reply:
Yep, caching the whole block will use too much memory.